### PR TITLE
Add travis to project

### PR DESCRIPTION
**Pull request by rmetzger**

### CODE REVIEW

**Comment by rmetzger** [comment]
I activated travis for the repository as well.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: java
+
+jdk:
+  - oraclejdk8
+  - oraclejdk7
+  - openjdk6
+
+script: mvn clean install
```
Comment (mxm) [inline .travis.yml]: Cascading-Flink doesn't pass all Cascading tests yet. We need to exclude some test cases (e.g. outer joins) for the Travis build to pass.

Comment (rmetzger) [inline .travis.yml]: I see. I'll make the travis builds work.

**Comment by rmetzger** [comment]
The build is passing even though tests are failing

```
Results :

Failed tests: 
  JoinFieldedPipesPlatformTest.testJoinMergeGroupBy:1943 planner should throw error on plan

Tests in error: 
  JoinFieldedPipesPlatformTest.testJoinOuterInner:739->handleJoins:768 » Planner
  JoinFieldedPipesPlatformTest.testJoinFilteredBranch:263 » Planner [lower*upper...
  JoinFieldedPipesPlatformTest.testJoinMixed:844 » Planner [loweroffset*upper*lo...
  JoinFieldedPipesPlatformTest.testJoinOuter:565->handleJoins:768 » Planner [low...
  JoinFieldedPipesPlatformTest.testJoinInnerOuter:657->handleJoins:768 » Planner
  ParallelCascadePlatformTest.testCascadeRaceCondition » Cascade flow failed: ra...

Tests run: 553, Failures: 1, Errors: 6, Skipped: 1
```

**Comment by mxm** [comment]
Yes, that's intentionally, so we could run all Cascading tests to see which ones fail. Otherwise, it would fail on the first failed test. For a CI integration this doesn't make much sense though..

**Comment by rmetzger** [comment]
so what would you like to do?
Change the settings to that the builds fail & ignore the failing tests (by annotating them)?
Keep everything as it is and merge it?

File: src/main/java/com/dataartisans/flink/cascading/planner/FlinkPlanner.java [hunk 2]
```diff
@@ -51,12 +51,12 @@
...
 		super();
 		this.classPath = classPath;
-
+		env.getConfig().disableSysoutLogging();
 		if (env.getParallelism() <= 0) {
 			// load the default parallelism from config
 			GlobalConfiguration.loadConfiguration(new File(CliFrontend.getConfigurationDirectoryFromEnv()).getAbsolutePath());
 			org.apache.flink.configuration.Configuration configuration = GlobalConfiguration.getConfiguration();
-			int parallelism = configuration.getInteger(ConfigConstants.DEFAULT_PARALLELISM_KEY_OLD, -1);
+			int parallelism = configuration.getInteger(ConfigConstants.DEFAULT_PARALLELISM_KEY, -1);
```
Comment (mxm) [inline src/main/java/com/dataartisans/flink/cascading/planner/FlinkPlanner.java]: No, this should have been `DEFAULT_PARALLELISM_KEY_OLD` but in the meantime the entry has been removed anyways :)

File: tools/log4j-travis.properties [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+log4j.appender.console.layout.ConversionPattern=%d{HH:mm:ss,SSS} %-5p %-60c %x - %m%n
+
+# -----------------------------------------------------------------------------
+# File (use 'file')
+# -----------------------------------------------------------------------------
+log4j.appender.file=org.apache.log4j.FileAppender
+log4j.appender.file.file=build.log
+log4j.appender.file.append=true
+log4j.appender.file.layout=org.apache.log4j.PatternLayout
+log4j.appender.file.layout.ConversionPattern=%d{HH:mm:ss,SSS} %-5p %-60c %x - %m%n
```
Comment (mxm) [inline tools/log4j-travis.properties]: Good fix.

**Comment by mxm** [comment]
Thanks for your pull request. I'd like to merge the Maven support first. Unfortunately, we can't annotate the tests because they are part of Cascading. We can, however, exclude certain classes which might be ok for Travis. It would be better if we could exclude certain methods because only a small number of tests fail.

**Comment by mxm** [comment]
I've merged your changes with a few adjustments in c550f4ad6e76c18cc9610f4723d2bbf2d7663b0b. Thank you!

**Pull request closed without merging**